src/mono/mono/mini/interp/transform.c

```diff
@@ -5742,23 +5487,7 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			}
 
 			int ret_mt = mint_type (m_class_get_byval_arg (klass));
-			if (mono_class_is_magic_int (klass) || mono_class_is_magic_float (klass)) {
-				g_assert (csignature->param_count == 1);
-#if SIZEOF_VOID_P == 8
-				if (mono_class_is_magic_int (klass) && td->sp [-1].type == STACK_TYPE_I4)
-					interp_add_conv (td, td->sp - 1, NULL, stack_type [ret_mt], MINT_CONV_I8_I4);
-				else if (mono_class_is_magic_float (klass) && td->sp [-1].type == STACK_TYPE_R4)
-					interp_add_conv (td, td->sp - 1, NULL, stack_type [ret_mt], MINT_CONV_R8_R4);
-#endif
-			} else if (klass == mono_defaults.int_class && csignature->param_count == 1)  {
```

Comment:
The removal of this seems unrelated to the rest of the change. Is it intentional ?

Reply:
@BrzVlad oops you're right, I meant to just kill the first if, not this one. Thanks!